datastore/datastore/storagebackend/postgresql/putobservations.go

```diff
@@ -104,6 +77,26 @@ func getTSColVals(tsMdata *datastore.TSMetadata) ([]interface{}, error) {
 		}
 	}
 
+	// --- END non-string metadata ---------------------------
+
+	// --- BEGIN string metadata (handleable with reflection) ---------------------------
+
+	rv := reflect.ValueOf(tsMdata)
+	for _, field := range reflect.VisibleFields(reflect.TypeOf(datastore.TSMetadata{})) {
+		methodName := fmt.Sprintf("Get%s", field.Name)
+		method := rv.MethodByName(methodName)
+		if field.IsExported() && (field.Type.Kind() == reflect.String) && (method.IsValid()) {
+			val, ok := method.Call([]reflect.Value{})[0].Interface().(string)
+			if !ok {
+				return nil, fmt.Errorf(
+					"method.Call() failed for method %s; failed to return string", methodName)
+			}
+			colVals = append(colVals, val)
+		}
+	}
```

Comment:
I was wondering if there is a way to do all the reflection upfront, and don't do it at runtime (as we are already somewhat limited on insertion speed... but that might very well not be the go code, but the DB)...

But anyway, if you want to go down the rabbit hole with me...

Go has a thing called _method expression_, with which you can call the method providing the receiver as a argument. So instead of `p.Distance(q)` you write:
```
distance := Point.Distance
distance(p, q)
```

If a _method expression_ could be derived using reflection for each of the metadata fields, we could do that upfront, and then simply call those in a loop.
But I haven't been able to find an example of how to do this. Maybe it is not possible, and if it is, maybe the performance is just as bad.

Reply:
I'll bring over my write load test from the other branch, and take the current code for a spin.

Reply:
Running my "writer" load test (from the other PR) on your branch, I don't see any problems. It runs fine, not errors in the first hour. I don't see a significant higher load (it might be 10% higher, hard to tell by just watching the numbers.

For the read-only load (I compared Github actions for your and my branch), they are pretty much the same.

May the conclusion is that even if reflection is slower, it is still not relevant compared to the DB performance?

Reply:
I think this should be feasible, and also save a bit of time by doing certain reflection operations once and for all in init(). How much time I'm like you not sure of, but I'll commit a change for it, so we can easily switch between the two implementations if we like.

Reply:
Made a change for this now. Though not entirely sure it was what you had in mind since it was a quite straightforward refactoring that didn't really introduce anything new except from moving some reflection calls to init().